buildprocess/configureRspack.js

```diff
@@ -57,19 +58,48 @@ function configureWebpack({
   config.module = config.module || {};
   config.module.rules = config.module.rules || [];
 
-  babelLoader = babelLoader || defaultBabelLoader(devMode);
-  // Use Babel to compile our JavaScript and TypeScript files.
+  // Use SWC to compile our JavaScript and TypeScript files.
   config.module.rules.push({
-    test: /\.(ts|js)x?$/,
+    test: /\.[jt]sx?$/,
     include: [
       path.resolve(terriaJSBasePath, "node_modules", "commander"),
       path.resolve(terriaJSBasePath, "lib"),
       path.resolve(terriaJSBasePath, "test"),
       path.resolve(terriaJSBasePath, "buildprocess", "generateDocs.ts"),
       path.resolve(terriaJSBasePath, "buildprocess", "generateCatalogIndex.ts"),
-      path.resolve(terriaJSBasePath, "buildprocess", "patchNetworkRequests.ts")
+      path.resolve(terriaJSBasePath, "buildprocess", "patchNetworkRequests.ts"),
+      path.resolve(cesiumDir, "Source")
     ],
-    use: [babelLoader]
+    exclude: [/[\\/]node_modules[\\/]/],
+    loader: "builtin:swc-loader",
+    /** @type {import('@rspack/core').SwcLoaderOptions} */
+    options: {
+      jsc: {
+        parser: {
+          syntax: "typescript",
+          decorators: true,
+          tsx: true
+        },
+        externalHelpers: true,
+        transform: {
+          // For MobX, see https://mobx.js.org/enabling-decorators.html
+          legacyDecorator: true,
+          react: {
+            runtime: "automatic",
+            development: devMode,
+            // FIXME: setting refresh to devMode fails the tests with:
+            // ReferenceError: $RefreshReg$ is not defined
+            refresh: false
+          }
+        }
+      },
+      env: {
+        coreJs: "3",
+        mode: "usage",
+        targets: ["since 2020-01-01 and not dead"]
```

Comment:
I don't know how old browsers that are currently supported, so this line needs to be adjusted.